gap/Bialgebroids.gi

```diff
@@ -528,7 +528,7 @@ InstallMethod( MorphismInAlgebroid,
             
             if not ( Source( l ) = UnderlyingVertex( S ) and
                      Target( l ) = UnderlyingVertex( T ) ) then
-                Error( "the path ", path, " is neither zero nor does it match the given source S or target T\n" );
```

Comment:
There are three conditions involved here:

- `IsZero( path )`
- `Source( l ) = UnderlyingVertex( S )`
- `Target( l ) = UnderlyingVertex( T )`

What are a and b?

Reply:
a = `Source( l ) = UnderlyingVertex( S )` and b = `Target( l ) = UnderlyingVertex( T )`

Reply:
We didn't finish the discussion yet :)

Reply:
With c = `IsZero( path )` the error is thrown if not(c) and not(a and b).

This is rather a linguistic question. A simplified version would be:

Does in English "neither c nor a or b" mean

* "not(c) and not(a or b)" 
or 
* "not(c) and (not(a) or not(b))"?

I guess in general both are possible interpretations. Here the situation is however more special as in "nor does it match the given source S or target T" the pronoun "it" belongs to both conditions a and b, which I think enforces the interpretation "not(a or b)". Therefore I prefer "and" over "or" here.

However it would probably be better to completely rewrite the error message. Suggestion:

`Error("the path ", path, " should either be zero or it should match the given source S and target T. This is not the case.\n");`

Reply:
Another suggestion: We should split the error message per if statement. What do you think?